R/get_latest_resource.R

```diff
@@ -0,0 +1,92 @@
+#' For these it is useful to be able to retrieve the latest
+#' resource. As of 5.7.2024 these data sets include:
+#' * gp-practice-populations
+#' * gp-practice-contact-details-and-list-sizes
+#' * nhsscotland-payments-to-general-practice
+#' * dental-practices-and-patient-registrations
+#' * general-practitioner-contact-details
+#' * prescribed-dispensed
+#' * prescriptions-in-the-community
+#' * community-pharmacy-contractor-activity
+#'
+#' @param dataset_name name of the dataset as found on
+#' \href{https://www.opendata.nhs.scot/}{NHS Open Data platform}
+#' @param rows (optional) specify the max number of rows to return.
+#' @param row_filters (optional) a named list or vector that specifies values of
+#'  columns/fields to keep.
+#' e.g. list(Date = 20220216, Sex = "Female").
+#' @param col_select (optional) a character vector containing the names of
+#' desired columns/fields.
+#' e.g. c("Date", "Sex").
+#' @param include_context (optional) If `TRUE` additional information about the
+#' resource will be added as columns to the data, including the resource ID, the
+#' resource name, the creation date and the last modified/updated date.
+#'
+#' @return a [tibble][tibble::tibble-package] with the data
+#' @export
+#'
+#' @examples
+#' dataset_name <- "gp-practice-contact-details-and-list-sizes"
+#'
+#' data <- get_latest_resource(dataset_name)
+#'
+#' filters <- list("Postcode" = "DD11 1ES")
+#' wanted_cols <- c("PracticeCode", "Postcode", "Dispensing")
+#'
+#' filtered_data <- get_latest_resource(
+#'   dataset_name = dataset_name,
+#'   row_filters = filters,
+#'   col_select = wanted_cols
+#' )
+#'
+get_latest_resource <- function(dataset_name,
+                                rows = NULL,
+                                row_filters = NULL,
+                                col_select = NULL,
+                                include_context = FALSE) {
+  applicable_datasets <- c(
+    "gp-practice-populations", "gp-practice-contact-details-and-list-sizes",
+    "nhsscotland-payments-to-general-practice", "dental-practices-and-patient-registrations",
+    "general-practitioner-contact-details", "prescribed-dispensed",
+    "prescriptions-in-the-community", "community-pharmacy-contractor-activity"
+  )
+
+
+
+
+  # check if data set is within applicable datasets
+  # throw error if not
+  if (!dataset_name %in% applicable_datasets) {
+    cli::cli_abort(
+      c(
+        "The dataset name supplied {.var {dataset_name}} is not within the applicable datasets.
+      These are:\n
+      {.var {applicable_datasets}}",
+        "x" = "Please see get_latest_reource documentation.",
+        "i" = "You can find dataset names in the URL
+      of a dataset's page on {.url www.opendata.nhs.scot}."
+      ),
+      call = rlang::caller_env()
+    )
+  }
+
+
+  # get the latest resource id
+  id <- get_latest_resource_id(dataset_name)
+
+  data <- get_resource(
+    id,
+    rows,
+    row_filters,
+    col_select,
+    include_context
```

Comment:
Naming the params protects us from weird errors if for some reason we re-order them in the future.
```suggestion
    res_id = id,
    rows = rows,
    row_filters = row_filters,
    col_select = col_select,
    include_context = include_context
```